### ISSUE TITLE
problems autosolving in 1.4.3

### ISSUE THREAD

**Issue by somokol** [Citrinate/gleamSolver#5]
Ever since I updated to the latest version, 1.4.3, it wont autosolve twitter follows. it throws a red bar stating i need to be logged in, though if I do a manual solve, I get an entry.

It also wont solve facebook visits, which it did before. 

Is there a way I can go back to 1.4.1

Anyhow, thanks for a great working script, I prefer it over the other one since it doesn’t popup every page it's working to solve.

**Comment by Citrinate** [comment]
I was able to identify and fix a Twitter follow bug in v1.4.4.

I couldn't replicate any issues with Facebook visits.  Sometimes though when you click on an entry, it will ask for additional information (your name, your birthday,...).  Previous versions of the script ignored this request and completed the entry anyway, which may raise suspicions and lead to detection.

If the script is refusing to solve an entry for this reason, you'll need to click on that entry and provide the necessary information before continuing.  I'll see what I can do to make it clear that this is the reason why the script refuses to continue.

You can downgrade to 1.4.1 [using this link](https://github.com/Citrinate/gleamSolver/raw/ec5c8fe3e376420895bcf56cad8bdf3707361987/gleamSolver.user.js).  You'll also need to disable auto-updates in your Tampermonkey/Greasemonkey settings

**Comment by somokol** [comment]
Thanks for the quick fix, though still not working with Twitter follows for me.

This is a contest I was trying it on... https://gleam.io/MYcdT/win-a-slickwrapped-xbox-one-or-playstation-4-

screencap of the error
http://imgur.com/Ut0kiZk

**Comment by Citrinate** [comment]
So are you actually logged into Twitter?  The script won't even attempt to complete the entry if you're not (ever since 1.4.1).  You can avoid this check by switching to "Raffle mode".

I also didn't design Gleam.solver and Gleam.helper to be used together, though I'm not sure if that's the cause of your issues.

**Comment by somokol** [comment]
I'm on 1.4.5, and tried Raffle Mode, it still wont auto solve the twitter follow, it gets stuck on the screen I show in the screen capture I linked.

http://i.imgur.com/J0ZXRZi.png

**Comment by Citrinate** [comment]
The script UI that shows the version # should re-appear after it attempts to complete all entries, but in your screenshot that's not happening.  I've been able to replicate that issue in Firefox, and I'll see if I can get that fixed in the hope that it's somehow related to the Twitter follows bug.  I haven't been able to replicate any of these issues in Chrome.

I'd also recommend that you use either Gleam.solver or Gleam.helper, and not both.  They weren't really designed to be used together, but I don't believe that's whats causing this issue.

In the meantime the link for v1.4.0 is here: https://github.com/Citrinate/gleamSolver/raw/4e801c7c23ef8a46df9b30be406e3b7f326b5810/gleamSolver.user.js

**Comment by Citrinate** [comment]
v1.4.6 fixes the issue I experienced on Firefox, but I don't think it will fix your main issue.

Looking at your second screenshot a bit closer though, I have to ask if you even have a Twitter account connected to your Gleam.io account?  I feel like you shouldn't see a "Connect Your Twitter Account" button if you do.

If it's possible to complete these entries without a Twitter account, then that's news to me.

I noticed an issue when I was working on v1.4.1, that when the script completed a Twitter entry, the activity sometimes wouldn't show up on my Twitter account (no follows, no tweets, no retweets).  I fixed this because I feared it was unintentional and may cause detections.  That's probably the change I made in v1.4.1 that broke things for you.

If you give me some more details I'll try to restore the ability to complete Twitter entries without Twitter accounts.

**Comment by somokol** [comment]
No, I don't have a Twitter account connected, yet I can still get an entry simply manually clicking the twitter follow button. I've noticed Gleam doesn’t verify certain tasks, probably because they cant, since the API of certain social media sites don’t offer such an ability for verification.

In version 1.4.0, it auto solved Twitter follows flawlessly, even though I didn’t have a Twitter account connected.

Can you re-link 1.4.0? Thanks.

**Comment by somokol** [comment]
I just noticed something, which might be why you cant replicate the problem, I posted a screenshot explanation...

http://imgur.com/a/GUspZ

**Comment by Citrinate** [comment]
Still not 100% sure how Gleam treats Twitter entries without a Twitter account.  I think on the creator's side there might be an option for them to require a Twitter account that not everyone uses.

I created a new account and was able to reproduce and fix the issue in some giveaways.  For others I tested it wouldn't even let me complete Twitter follows manually, and some even had a mix of ones I could complete and ones I couldn't.

Not sure if this is normal for you, but the script v1.4.7 should now work exactly as it did in v1.4.0 with respect to Twitter entries without an account.
